contracts/modules/andromeda-schema/src/query.rs

```diff
@@ -0,0 +1,29 @@
+use andromeda_modules::schema::ValidateDataResponse;
+use andromeda_std::error::ContractError;
+use cosmwasm_std::Storage;
+use jsonschema_valid::{schemas, Config};
+use serde_json::{from_str, json};
+
+use crate::state::SCHEMA;
+
+pub fn validate_data(
+    storage: &dyn Storage,
+    data: String,
+) -> Result<ValidateDataResponse, ContractError> {
+    let schema = json!(SCHEMA.load(storage)?);
+    let data_instance = from_str(data.as_str()).map_err(|_| ContractError::CustomError {
+        msg: "Invalid data JSON".to_string(),
+    })?;
+
+    let config = Config::from_schema(&schema, Some(schemas::Draft::Draft7)).map_err(|_| {
+        ContractError::CustomError {
+            msg: "Validation Error".to_string(),
+        }
+    })?;
+
+    if config.validate(&data_instance).is_ok() {
+        Ok(ValidateDataResponse { is_valid: true })
```

Comment:
```suggestion
        Ok(ValidateDataResponse { is_valid: config.validate(&data_instance).is_ok() })
```

Remove the if/else statement

Reply:
I think we're actually losing a potential error here. We can tell the user if the data is valid or not but we can't tell them what's wrong with it when it isn't valid.

Reply:
For this, we can use `Result<(), ContractError>` return value instead of true/false...
In your opinion?

Reply:
I think we can return a response but yes we should bubble up any errors from `config.validate`.

Reply:
Resolved this.